src/main/sc_client.c

```diff
@@ -0,0 +1,127 @@
+/*
+ * secrets.c
+ *
+ * Copyright (C) 2023 Aerospike, Inc.
+ *
+ * All rights reserved.
+ *
+ * THIS IS UNPUBLISHED PROPRIETARY SOURCE CODE. THE COPYRIGHT NOTICE ABOVE DOES
+ * NOT EVIDENCE ANY ACTUAL OR INTENDED PUBLICATION.
+ */
+
+//==========================================================
+// Includes.
+//
+
+#include "sc_secrets.h"
+#include "sc_socket.h"
+#include "sc_logging.h"
+#include "sc_client.h"
+#include "sc_error.h"
+
+#include <arpa/inet.h>
+#include <errno.h>
+#include <unistd.h>
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/socket.h>
+#include <netinet/in.h>
+
+#include "jansson.h"
+
+//==========================================================
+// Public API.
+//
+
+sc_client*
+sc_client_init(sc_client* c, sc_cfg* cfg) {
+	c->cfg = cfg;
+	return c;
+}
+
+sc_client*
+sc_client_new(sc_cfg* cfg) {
+	sc_client* c = (sc_client*) malloc(sizeof(sc_client));
+	return sc_client_init(c, cfg);
+}
+
+sc_err
+sc_secret_get_bytes(const sc_client* c, const char* path, uint8_t** r, size_t* size_r) {
+	sc_err err = { 0 };
+    sc_cfg* cfg = c->cfg;
+
+	// path format will be "secrets[:resource_substring]:key"
+	const char* suffix = path + sizeof(SC_SECRETS_PATH_REFIX) - 1;
+	uint32_t suffix_len = (uint32_t)strlen(suffix);
+
+	if (suffix_len == 0) {
+		sc_g_log_function("ERR: empty secret key");
+		err.code = SC_FAILED_INTERNAL;
+		return err;
+	}
+
+	const char* res = NULL;
+	uint32_t res_len = 0;
+	const char* key = strrchr(suffix, ':');
+
+	if (key == NULL) {
+		// no resource name
+		key = suffix;
+	}
+	else {
+		res = suffix;
+		res_len = (uint32_t)(key - suffix);
+		key++;
+	}
```

Comment:
review this logic

Reply:
This parses the optional resource name, `secrets[:<resource_name>]:<secret_key>`